activesupport/lib/active_support/core_ext/string/output_safety.rb

```diff
@@ -261,8 +261,14 @@ def %(args)
       self.class.new(super(escaped_args))
     end
 
-    def html_safe?
-      defined?(@html_safe) && @html_safe
+    if RUBY_VERSION < "3.0"
+      def html_safe?
+        defined?(@html_safe) && @html_safe
+      end
+    else
+      def html_safe?
+        @html_safe
+      end
```

Comment:
Maybe silly, but one way to handle this without warning on all rubies and be a tiny bit faster:

```suggestion
  attr_reader :html_safe
  alias_method :html_safe?, :html_safe
  undef_method :html_safe
```

But feel free to disregard, your change is fine by me.